Automattic/jetpack
# Jetpack Settings: Add loading/error messages while fetching rewind state

On Jetpack settings page, we display a default message under `Backups and security scanning` saying something similar to `You have paid for backups but they're not yet active.` and specifying that the user needs to click on a nonexistent CTA to perform an action.

After digging a bit, we discover that the message is being displayed while fetching rewind state and also, when we have an error on that API call.

The idea of this PR is to fix this behavior, and add a loading message when fetching the rewind state, and an error message when it throws an error, to prevent displaying wrong messages to customers.






## Proposed changes:

* Add `isFetchingRewindStatus` selector to `LoadingCard` component
* Add `Checking site status…` message when loading rewind status.
* Add `The Jetpack Backup and Scan status could not be retrieved at this time.` message when it fails to fetch the rewind status.

### Demo

#### Before
https://github.com/Automattic/jetpack/assets/1488641/ca21321f-bc3e-4caf-a412-ff121219ca4b

#### After
https://github.com/Automattic/jetpack/assets/1488641/4c5b07b9-2c1c-4afb-9f7b-457e3d177bfe

### Other information:


## Jetpack product discussion


1202586245449203-as-1204562823140386/f

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a Jurassic Ninja site with this branch: https://jurassic.ninja/create?jetpack-beta&branches.jetpack=fix/backup-security-settings-cta&wp-debug-log
* Add Jetpack Security plan.
  * You could navigate to Jetpack > My Jetpack, then click on `Purchase` under VaultPress Backup.
  * Under the interstitial page, click on `Get Security` (right side) and complete the checkout process.
* Navigate to Jetpack > Settings and you should see it behaves similar to the `After` demo video posted above.
